backend/src/controllers/newsletterController.js

```diff
@@ -0,0 +1,89 @@
+const { v4: uuidv4, validate } = require("uuid");
+
+const { isValidEmail } = require("../utils/validation");
+
+const { insertOne, findOne, updateOne } = require("../services/db");
+const { sendEmail } = require("../services/email");
+
+async function register(req, res) {
+  const { email } = req.body;
+
+  if (!email) {
+    res.status(400).send({ message: "Send a email address" });
+    return;
+  }
+
+  if (!isValidEmail(email)) {
+    res.status(400).send({ message: "Send a valid email address" });
+    return;
+  }
+
+  const existsEmail = await findOne("newsletters", { email });
+
+  if (existsEmail) {
+    res.status(400).send({
+      message: "Send another email address, this email address already exists",
+    });
+    return;
+  }
+
+  try {
+    const newSub = { email, status: false, confirmationToken: uuidv4() };
+
+    await insertOne("newsletters", newSub);
+
+    let message = {
+      from: "newsletter@juntavagas.com.br",
+      to: email,
+      subject: "Confirme seu e-mail",
+      html: `<h1>Confirme seu e-mail</h1>
+      <p>Por favor, ative seu e-mail no link abaixo para receber nosso conteúdo.</p>
+      <a href="http://localhost:3000/newsletter/emailConfirmation/${newSub.confirmationToken}" target="_blank">Confirme seu e-mail</a>`,
```

Comment:
Um sugestão aqui é criar uma variável de ambiente no arquivo **.env** chamada por exemplo **URL_NEWSLETTER** com o valor **http://localhost:3000/newsletter** .

Ai na linha 41 colocar a variável de ambiente **URL_NEWSLETTER** no lugar de http://localhost:3000/newsletter , pois dessa forma quando a aplicação for colocada em ambiente de teste ou produção apenas precisa alterar no arquivo **.env** evitando ter que alterar o código

Reply:
Otima sugestão, nem tinha notado isso.